src_c/base.c

```diff
@@ -192,6 +192,34 @@ pg_CheckSDLVersions(void) /*compare compiled to linked*/
                      linked.major, linked.minor, linked.patch);
         return 0;
     }
+    else if (linked.major == 2 && linked.minor == 0 && linked.patch < 14 &&
+             compiled.patch >=
+                 14) {  // major and minor versions match, check edge case
+        /* SDL 2.0.14 replaces some macros with symbols, see
+         * https://github.com/libsdl-org/SDL/commit/316ff3847b4d9d87d9b0aab15321461db0e8ae0b
+         */
+        PyErr_Format(PyExc_RuntimeError,
+                     "Known SDL incompatibility detected! (compiled with "
+                     "version %d.%d.%d, linked to %d.%d.%d)",
+                     compiled.major, compiled.minor, compiled.patch,
+                     linked.major, linked.minor, linked.patch);
+        return 0;
+    }
+    else if ((linked.major == compiled.major &&
+              linked.minor == compiled.minor &&
+              linked.patch < compiled.patch) ||
+             (linked.major == compiled.major &&
+              linked.minor < compiled.minor)) {
```

Comment:
These two lines can probably be removed, `linked.minor != compiled.minor` check happens in the first `if` block

Reply:
This was just for clarity.

Reply:
This is something the compiler will optimise, unless you pass -O0, and I think it's more readable like this. If you think the other way round is more readable, then by all means go ahead! If you are concerned about performance, this is just a one-off check using a few cycles, I wouldn't worry about it!

Reply:
I was not worried about performance, but more about readability. But eh, anyways it's a minor thing, should be fine